subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/problems/JsonModelWriter.kt

```diff
@@ -246,8 +246,12 @@ class JsonModelWriter(val writer: Writer) {
 
     private
     fun jsonString(value: String) {
-        buffer.addJsonEscapedString(value)
-        write(buffer.toStringAndRecycle())
+        if (value.isEmpty()) {
```

Comment:
Sounds like a bug in `CharBuf` if you ask me.

https://github.com/apache/groovy/blob/904841373d4a4be04e2a1c675bde4b21baaef6b6/subprojects/groovy-json/src/main/java/org/apache/groovy/json/internal/CharBuf.java#L380

(or maybe that choice makes sense in the use cases it was meant for)

Reply:
Yeah, maybe one is supposed to check the result of `toStringAndRecycle()` 🤷